### ISSUE TITLE
Feature request - add `orUndefined` to complement `orNull`

### ISSUE THREAD

**Issue by Juraj-Masiar** [shogogg/ts-option#15]
I know Scala doesn't have `orUndefined`, but in JavaScript it feels more useful than `orNull`.

Very common use case would be supplying value for the default parameter (which are used only if you omit the value or supply `undefined`:
```
function fn(a: number, b: number, c = 'hello') {
}
const strOpt = some('abc');
fn(1, 2, strOpt.getOrElseValue(undefined));  // this won't compile!
fn(1, 2, strOpt.orNull);  // this won't compile or work!
fn(1, 2, strOpt.orUndefined);  // this would work!
```

**Comment by shogogg** [comment]
@Juraj-Masiar The idea sounds good for me! So it's implemented at v2.1.0.

Sorry for making you wait, but this release is published on npm soon.

Thank for your feature request.

**Comment by Juraj-Masiar** [comment]
**Thank you for implementing this!**
I was thinking about this particular issue yesterday evening before sleeping and now it's implemented? Or am I still sleeping? :D

I'm looking forward to refactore my code with this! Thanks again!
